api/graphql/types/reservation/filtersets.py

```diff
@@ -145,11 +146,12 @@ def get_only_with_permission(self, qs: QuerySet, name: str, value: bool) -> Quer
         if not value:
             return qs
 
-        user = self.request.user
-        viewable_units = get_units_where_can_view_reservations(user)
+        user: AnyUser = self.request.user
         if user.is_anonymous:
             return qs.none()
-        return qs.filter(Q(reservation_unit__unit__in=viewable_units) | Q(user=user)).distinct()
+
+        viewable_units = get_units_where_can_view_reservations(user)
+        return qs.filter(Q(reservation_unit__unit__in=viewable_units))
```

Comment:
Improved performance on this and a bunch of other similar queries by using the request user's unit and unit group information to filter instead of using the helper methods. Managed to remove all helper methods that find units like this. Re-reviews are in order @ranta @vergama